Newspack/Newspack/Folders/FolderManager.swift

```diff
@@ -0,0 +1,143 @@
+import Foundation
+
+/// FolderManager provides a simplified interface for working with the file system.
+/// Operations are conducted relative to the current working folder.
+///
+class FolderManager {
+
+    // An instance of a FileManager
+    private let fileManager: FileManager
+
+    // All relative paths are mapped beneath thie specified rootFolder.
+    private let rootFolder: URL
+
+    // The current working folder. This will be either the rootFolder or
+    // one of its children.
+    private(set) var currentFolder: URL
+
+    /// A convenience method to create a new temporary directory and returns its URL.
+    ///
+    /// - Returns: A file url to the newly created temporary directory, or nil
+    /// if the directory could not be created.
+    ///
+    static func createTemporaryDirectory() -> URL? {
+        let documentDirectory = FileManager.default.urls(for: .documentDirectory, in: .userDomainMask).first
+        return try? FileManager.default.url(for: .itemReplacementDirectory, in: .userDomainMask, appropriateFor: documentDirectory, create: true)
+    }
+
+    /// Initializes the FolderManager, optionally specifying its default
+    /// rootFolder.
+    ///
+    /// - Parameter rootFolder: Optional. A URL to an existing folder to
+    /// use as the root of any relative paths. The folder must be writable.
+    /// If not specified the default rootFolder is the document directory.
+    ///
+    init(rootFolder: URL? = nil) {
+        fileManager = FileManager()
+
+        guard let documentDirectory  = fileManager.urls(for: .documentDirectory, in: .userDomainMask).first else {
+            fatalError()
+        }
+
+        var isDirectory: ObjCBool = false
+        if
+            let root = rootFolder,
+            fileManager.fileExists(atPath: root.path, isDirectory: &isDirectory),
+            isDirectory.boolValue,
```

Comment:
I'm tempted to say `folderExists` could be used here... *but* yeah it would have to be a static method, and that's not cool. 

Another alternative (I think) would be to mark this initializer as convenience, which should allow for method calls. But... yeah probably not worth it!!